include/boost/url/detail/impl/pct_encoding.ipp

```diff
@@ -81,6 +81,106 @@ key_equal_encoded(
     }
 }
 
+int
+pct_decode_compare_unchecked(
+    string_view lhs,
+    string_view rhs) noexcept
+{
+    auto consume_one = []
+        (string_view::iterator& it,
+         char &c,
+         std::size_t& n)
+    {
+        if(*it != '%')
+        {
+            c = *it;
+            ++it;
+        }
+        else
+        {
+            pct_decode_unchecked(
+                &c,
+                &c + 1,
+                string_view(it, 3));
+            it += 3;
+        }
+        ++n;
+    };
+
+    std::size_t lhs_n = 0;
+    std::size_t rhs_n = 0;
+    auto lhs_it = lhs.begin();
+    auto rhs_it = rhs.begin();
+    char lhs_c = '\0';
+    char rhs_c = '\0';
```

Comment:
instead of all this lhs rhs nonsense which can hypnotize you with its repetition and slight dissimilarities, how about these names;
```
n0, n1, it0, it1, c0, c1
```